### PR TITLE
Add utility to manipulate files en masse 

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
**Description**

This PR adds the ability to pass in a configuration and manipulate files in bulk. Currently it just works for `copy` and `mkdir`.

A sample YAML file:
```
mkdir:
- ${CRMTEST}/output
copy:
- - ${CRMTEST}/fakefile1
  - ${CRMTEST}/output/fakefile1
- - ${CRMTEST}/fakefile2
  - ${CRMTEST}/output/fakefile2
```

To use it in a python script:
```
from pygw.utils import FileHandler
FileHandler(path='/export/emc-lw-cmartin/cmartin/work/tmp/test.yaml').sync()
```

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update (eventually!)

**How Has This Been Tested?**

- [x] able to copy files and make directories on RHEL8 GFE
-->
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

**Review by RussTreadon-NOAA — APPROVED** [review]
Seems ok but I have not tested.

File: ush/python/pygw/src/pygw/utils.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    """Function to make all directories specified in the list
+
+    Parameters
+    ----------
+    dirlist : list
+              List of directories to create
+    """
+    for d in dirlist:
+        mkdir(d)
+        print(f'Created {d}') # TODO use logger
```
Comment (aerorahul) [inline ush/python/pygw/src/pygw/utils.py]: Generally try not to use single character variables.
```suggestion
    for dd in dirlist:
        mkdir(dd)
        print(f'Created {dd}') # TODO use logger
```

File: ush/python/pygw/src/pygw/utils.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+from .yaml_file import YAMLFile
+from .fsutils import cp, mkdir
+
+__all__ = ['FileHandler']
+
+
+class FileHandler():
```
Comment (aerorahul) [inline ush/python/pygw/src/pygw/utils.py]:
```suggestion
class FileHandler:
```

File: ush/python/pygw/src/pygw/utils.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        if path and config:
+            raise Exception('Both path and config are defined. Only one can be.')
+        elif path:
+            # read in the YAML file for the configuration
+            self.config = YAMLFile(path=path)
+        elif config:
+            self.config = config
+        else:
+            raise Exception('Neither path nor config are defined. One needs to be.')
```
Comment (aerorahul) [inline ush/python/pygw/src/pygw/utils.py]:
```suggestion
            raise Exception("Neither 'path' nor 'config' are defined. One needs to be.")
```

**Review by aerorahul — COMMENTED** [review]
Can you please provide an example as `example.yaml` and a few files `a.txt`, `b.txt` etc. that it can operate on?

**Comment by CoryMartin-NOAA** [comment]
@aerorahul an example in the repo somewhere or just attach them here?

**Comment by aerorahul** [comment]
> @aerorahul an example in the repo somewhere or just attach them here?

I can write the test.

**Comment by CoryMartin-NOAA** [comment]
[sample.zip](https://github.com/NOAA-EMC/global-workflow/files/10197159/sample.zip)
@aerorahul see attached, a sample YAML, a dummy file, and a py script that just needs the path modified to test.

**Review by aerorahul — APPROVED** [review]
includes tests with the feature.
looks good.

**Comment by CoryMartin-NOAA** [comment]
yes thank you @aerorahul for the test addition :-)

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Looks good. Might revisit the "list of lists" format for copy later.

**Pull request merged**